src/foam/u2/view/RichChoiceView.js

```diff
@@ -232,8 +232,11 @@ foam.CLASS({
         .addClass(this.myClass())
         .start()
           .addClass(this.myClass('selection-view'))
-          .on('click', function() {
-            self.isOpen_ = ! self.isOpen_;
+          .enableClass('disabled', this.mode$.map((mode) => mode === foam.u2.DisplayMode.DISABLED))
+          .callIf(this.mode === foam.u2.DisplayMode.RW, function() {
```

Comment:
There's a but here.

Using callIf like this means that if mode was DISABLED when we first rendered, then the click listener won't be hooked up, but if we later change mode to ENABLED, the click listener will still not be attached.  It's not dynamic.

Reply:
Yeah, I wasn't sure how to get around this. I know using an Action would let us use `isEnabled`, but the combination of trying to set the content of the action to the custom `selectionView` as well as dealing with overriding all of the default action styles made me want to avoid using an action. Should I bite the bullet and try to get it to work as an action or is there an easy alternative to using `callIf` the way it already is? I guess I could use a slot and conditionally put the on click listener in the callback, right?

Reply:
You could do that, but it seems simpler to just move your conditional logic into the click listener.  Rather than jump through hoops to prevent attaching the listener when we might not need it, just always attach the listener and add logic to your event handler to take appropriate action based upon the current state.

Reply:
Well that makes a lot more sense, haha. Thanks, updated.